File: examples/anthropic-service/src/background-path.ts
import { MemexAI } from "@memexai/sdk"

const memory = new MemexAI({
  url: process.env["MEMEX_URL"] ?? "http://localhost:8080",
  apiKey: process.env["MEMEX_API_KEY"] ?? "dev-agent-key",
}).forUser({
  userId: process.env["MEMEX_USER_ID"] ?? "anthropic_service_background_user",
  actor: "anthropic-service-background-path",
})

console.log("background_path: after the Anthropic Messages tool-use loop returns, extract durable facts")
const insight = "http_client permanent failure: the API limits request bodies to 256 KiB."
const result = await memory.remember({ text: insight, toolCallId: "anthropic_background_http_client" })

console.log(`learned_after_response: ${insight}`)
console.log(`writes: ${result.writes.map((write) => write.path).join(", ") || "none"}`)

File: examples/llamaindex-python/background_path.py
import asyncio
import os

from dotenv import load_dotenv
from memexai import MemexAI


async def main() -> None:
    load_dotenv()
    memex_url = os.getenv("MEMEX_URL") or f"http://localhost:{os.getenv('MEMEX_PORT', '8080')}"
    memex = MemexAI(url=memex_url, api_key=os.getenv("MEMEX_API_KEY", "dev-agent-key"))
    try:
        memory = memex.for_user(os.getenv("MEMEX_USER_ID", "llamaindex_background_user"), actor="llamaindex-background-path")
        print("background_path: after the LlamaIndex response, a workflow step extracts reusable facts")
        result = await memory.remember("LlamaIndex connector permanent failure: the source API caps page size at 100 items.")
        print(result.get("writes", []))
    finally:
        await memex.close()


if __name__ == "__main__":
    asyncio.run(main())
